invisible_cities/core/fit_functions.py

```diff
@@ -137,8 +137,8 @@ def fit(func, x, y, seed=(), fit_range=None, **kwargs):
     fitx = fitf(x)
     nonzero = fitx != 0
 
-    sigma_r = kwargs.get("sigma", None)
-    if sigma_r is not None:
+    sigma_r = kwargs.get("sigma", 1)
+    if np.size(sigma_r) is not 1:
```

Comment:
Mmm, I don't like this `size` check... How about setting the default to `np.ones` of the corresponding size and applying the `nonzero` selection over it?